apps/framework-cli/src/framework/typescript/ts_scripts/blocks.ts

```diff
@@ -0,0 +1,109 @@
+import process from "node:process";
+import { ClickHouseClient } from "@clickhouse/client-web";
+import fastq, { queueAsPromised } from "fastq";
+import { getClickhouseClient, walkDir, Blocks } from "@514labs/moose-lib";
+
+interface BlocksQueueTask {
+  chClient: ClickHouseClient;
+  blocks: Blocks;
+  retries: number;
+}
+
+class DependencyError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "DependencyError";
+  }
+}
+
+const [
+  ,
+  BLOCKS_DIR_PATH,
+  CLICKHOUSE_DB,
+  CLICKHOUSE_HOST,
+  CLICKHOUSE_PORT,
+  CLICKHOUSE_USERNAME,
+  CLICKHOUSE_PASSWORD,
+  CLICKHOUSE_USE_SSL,
+] = process.argv;
+
+const clickhouseConfig = {
+  username: CLICKHOUSE_USERNAME,
+  password: CLICKHOUSE_PASSWORD,
+  database: CLICKHOUSE_DB,
+  useSSL: CLICKHOUSE_USE_SSL,
+  host: CLICKHOUSE_HOST,
+  port: CLICKHOUSE_PORT,
+};
+
+const createBlocks = async (chClient: ClickHouseClient, blocks: Blocks) => {
+  for (const query of blocks.setup) {
+    try {
+      await chClient.command({ query });
+      console.log(`Created block using query ${query}`);
+    } catch (err) {
+      console.error(`Failed to create a block: ${err}`);
+
+      if (err && JSON.stringify(err).includes(`UNKNOWN_TABLE`)) {
+        throw new DependencyError(err.toString());
+      }
+    }
+  }
+};
+
+const deleteBlocks = async (chClient: ClickHouseClient, blocks: Blocks) => {
+  for (const query of blocks.teardown) {
+    try {
+      await chClient.command({ query });
+      console.log(`Deleted block using query ${query}`);
+    } catch (err) {
+      console.error(`Failed to delete block: ${err}`);
+    }
+  }
+};
+
+const asyncWorker = async (task: BlocksQueueTask) => {
+  await deleteBlocks(task.chClient, task.blocks);
+  await createBlocks(task.chClient, task.blocks);
+};
+
+const main = async () => {
+  const chClient = getClickhouseClient(clickhouseConfig);
+  console.log(`Connected`);
+
+  const blocksFiles = walkDir(BLOCKS_DIR_PATH, ".ts", []);
+  const numOfBlockFiles = blocksFiles.length;
+  console.log(`Found ${numOfBlockFiles} blocks files`);
+
+  const queue: queueAsPromised<BlocksQueueTask> = fastq.promise(asyncWorker, 1);
```

Comment:
Agreed, it would be great to not duplicate & maintain this over multiple languages. Would have to see if that means we have a process per file instead of one process to create all blocks.